platforms/software/reporting/src/main/java/org/gradle/api/reporting/internal/SimpleReport.java

```diff
@@ -28,29 +29,22 @@
 import java.io.File;
 
 public abstract class SimpleReport implements ConfigurableReport {
-    private final String name;
-    private final Describable displayName;
-    private final OutputType outputType;
 
     public SimpleReport(String name, Describable displayName, OutputType outputType) {
-        this.name = name;
-        this.displayName = displayName;
-        this.outputType = outputType;
+        this.getName().set(name);
+        this.getDisplayName().set(displayName.getDisplayName());
+        this.getOutputType().set(outputType);
     }
 
     @Override
-    public String getName() {
-        return name;
-    }
+    public abstract Property<String> getName();
```

Comment:
💅 Probably not that important since it's internal type, but should this be just a Provider?

Reply:
🤔 This means fields for providers and `Providers.of`, right?

Reply:
I guess we can use that, yes

Reply:
But we could also leave it I guess, since it's internal. I don't have strong preference

Reply:
Eh, maybe using Provider is safer :)